applications/incompressible_navier_stokes/navier_stokes_manufactured/application.h

```diff
@@ -0,0 +1,798 @@
+  double const                 lambda;
+  double const                 a;
+  double const                 n;
+};
+
+template<int dim>
+class RightHandSide : public dealii::Function<dim>
+{
+public:
+  RightHandSide(bool const     include_convective_term,
+                double const & normal_x,
+                double const & normal_y,
+                double const & nu_oo,
+                double const & nu_margin,
+                double const & kappa,
+                double const & lambda,
+                double const & a,
+                double const & n)
+    : dealii::Function<dim>(dim, 0.0),
+      include_convective_term(include_convective_term),
+      normal_x(normal_x),
+      normal_y(normal_y),
+      nu_oo(nu_oo),
+      nu_0(nu_margin - nu_oo),
+      kappa(kappa),
+      lambda(lambda),
+      a(a),
+      n(n)
+  {
+  }
+
+  double
+  value(dealii::Point<dim> const & p, unsigned int const component = 0) const
+  {
+    /*
+     * The solution laid out above is inserte into the momentum balance
+     * equation to derive the vector f on the right-hand side.
+     *
+     * f = d/dt(u) + c (grad(u)) * u + grad(p)
+     *     - nu * div(grad(u)) - 2 * sym_grad(u) * grad(nu)
+     */
+
+    double t      = this->get_time();
+    double x      = p[0];
+    double y      = p[1];
+    double sin_x  = std::sin(x);
+    double sin_y  = std::sin(y);
+    double cos_x  = std::cos(x);
+    double cos_y  = std::cos(y);
+    double sin_t  = std::sin(t);
+    double cos_t  = std::cos(t);
+    double sin_xy = std::sin(x * y);
+
+    double u1 = sin_x * cos_y * cos_t;
+    double u2 = -cos_t * cos_x * sin_y;
+
+    double du1_dt = -sin_x * cos_y * sin_t;
+    double du1_dx = cos_x * cos_y * cos_t;
+    double du1_dy = -sin_x * sin_y * cos_t;
+    double du2_dt = sin_t * cos_x * sin_y;
+    double du2_dx = cos_t * sin_x * sin_y;
+    double du2_dy = -cos_t * cos_x * cos_y;
+
+    double du1_dxx = -sin_x * cos_y * cos_t;
+    double du1_dyy = -sin_x * cos_y * cos_t;
+    double du2_dxx = cos_t * cos_x * sin_y;
+    double du2_dyy = cos_t * cos_x * sin_y;
+
+    double dp_dx = -sin_xy * cos_t * y;
+    double dp_dy = -sin_xy * cos_t * x;
+
+    dealii::Tensor<2, dim> grad_u;
+    grad_u[0][0] = du1_dx;
+    grad_u[0][1] = du1_dy;
+    grad_u[1][0] = du2_dx;
+    grad_u[1][1] = du2_dy;
+
+    dealii::Tensor<1, dim> div_grad_u;
+    div_grad_u[0] = du1_dxx + du1_dyy;
+    div_grad_u[1] = du2_dxx + du2_dyy;
+
+    // add time derivative terms
+    dealii::Tensor<1, dim> rhs;
+    rhs[0] += du1_dt;
+    rhs[1] += du2_dt;
+
+    // convective term
+    if(include_convective_term)
+    {
+      dealii::Tensor<1, dim> u;
+      u[0] = u1;
+      u[1] = u2;
+      rhs += grad_u * u;
+    }
+
+    // pressure gradient
+    rhs[0] += dp_dx;
+    rhs[1] += dp_dy;
+
+    // velocity Laplacian respecting the
+    // rheological law to obtain the kinematic viscosity
+    // nu_oo + (nu_0 - nu_oo) * [kappa + (lambda*shear_rate)^a]^[(n-1)/a]
```

Comment:
Personally, I would skip this comment, because the actual code is more or less the same.